train_model.py

```diff
@@ -163,9 +163,10 @@ def __len__(self):
     os.makedirs(newmodeldir, exist_ok=True)
 
     # ---- get model setup
-    checkpoint = torch.load(args.model)
-
+    # retrieve device from config.ini file
     device = get_torch_device(args.gpuid)
+    # map baseline model to desired device
+    checkpoint = torch.load(args.model, map_location = device)
```

Comment:
nice catch. in other code we use this approach

https://github.com/choosehappy/PytorchDigitalPathology/blob/d9925f3795c47112b6ccbffdc780b1e4ddd72145/segmentation_epistroma_unet/make_output_unet_cmd.py#L53

have you already seen it?

would this or that approach be more ideal?